### ISSUE TITLE
Warn on single-bracket conditionals in bash?

### ISSUE THREAD

**Issue by cxw42** [koalaman/shellcheck#1812]
Thanks for this tool, which I have been using for many years with good results!  This is my first issue (that I recall); please pardon any errors.  I have a feeling this must have been covered somewhere, but I can't find it in the wiki or issues.

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
foo=1
if [ -d $foo ]; then   # using test, not [[ ]]
echo yep
fi

```

#### Here's what shellcheck currently says:

Web:
```
$ shellcheck myscript
No issues detected!
```

Console (v0.4.6): no output; returns status 0

#### Here's what I wanted or expected to see:

Warning: use `[[ ]]` instead of `[ ]` in `bash` scripts

[wooledge](http://mywiki.wooledge.org/BashGuide/TestsAndConditionals#Conditional_Blocks_.28if.2C_test_and_.5B.5B.29) says:

> Good Practice:
> Whenever you're making a Bash script, you should always use [[ rather than [.

Thanks for considering this request!

**Comment by Nightfirecat** [comment]
Duplicate of #887

**Comment by cxw42** [comment]
Thanks!
